x/mongo/driver/topology/errors.go

```diff
@@ -15,13 +19,39 @@ type ConnectionError struct {
 
 // Error implements the error interface.
 func (e ConnectionError) Error() string {
+	message := e.message
+	if e.init {
+		message += "error occured during connection handshake"
```

Comment:
If `e.message` is non-empty, this will concatenate a string to it without any spaces. It only works right now because we aren't setting a message for errors encountered during `connection.connect()`. How about doing:

```
fullMsg := "error occured during connection handshake
if message != "" {
    fullMsg = fmt.Sprintf("%s: %s", fullMsg, message)
}

message = fullMsg
```

This way, we get both pieces of information and don't have weird spacing if `e.message` is empty.